src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -887,45 +921,39 @@ private void ExecuteBlockActions<TBlockAction>(
 
             foreach (var blockAction in blockActions)
             {
-                var codeBlockAction = blockAction as CodeBlockAnalyzerAction;
-                if (codeBlockAction != null)
+                if (blockAction is CodeBlockAnalyzerAction codeBlockAction)
                 {
                     var context = new CodeBlockAnalysisContext(declaredNode, declaredSymbol, semanticModel,
                         AnalyzerOptions, addDiagnostic, isSupportedDiagnostic, filterSpan, isGeneratedCode, cancellationToken);
 
                     ExecuteAndCatchIfThrows(
                         codeBlockAction.Analyzer,
-                        data => data.action(data.context),
-                        (action: codeBlockAction.Action, context: context),
+                        static data => data.action(data.context),
+                        argument: (action: codeBlockAction.Action, context),
```

Comment:
made lambdas explicitly static to make it clear they must not capture naything (since that would defeat the purpose).  my followup pr applies this to all the lambdas in this file.